test/unit/processing/process-payment-file.test.js

```diff
@@ -0,0 +1,63 @@
+const processPaymentFile = require('../../../app/processing/process-payment-file')
+
+jest.mock('../../../app/processing/batch')
+const batch = require('../../../app/processing/batch')
+
+jest.mock('../../../app/processing/quarantine-file')
+const quarantineFile = require('../../../app/processing/quarantine-file')
+
+jest.mock('../../../app/processing/reprocess-if-needed')
+const reprocessIfNeeded = require('../../../app/processing/reprocess-if-needed')
+
+jest.mock('../../../app/processing/download-and-parse')
+const downloadAndParse = require('../../../app/processing/download-and-parse')
+const { sfiPilot } = require('../../../app/schemes')
+
+global.console.error = jest.fn()
+global.console.log = jest.fn()
+
+let filename
+
+describe('Process payment file', () => {
+  beforeEach(() => {
+    filename = 'SITIELM0001_AP_20220317104956617.dat'
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  test('Should not process file again if previously processed', async () => {
```

Comment:
test names in this file don't tell me what to expect, e.g. "current sequence is greater than expected sequence and is ignored" this doesn't tell me that we're expecting `console.log` to be called nor does "Both current sequence and expected sequence are equal so download and parse" tell me we're calling some function named `downloadAndParse`

Reply:
Yeah I see what you mean.  I haven't changed those tests or functionality so didn't look at them.  I would be tempted to say this is out of scope of this ticket so not going to address it.  But having a test that's dependent on the order of console logging feels very wrong to me.  What if we add an extra console log in the process? the tests would fail but the behaviour would still be correct.

I think this should be addressed now so I'll look to see what can be done.

Reply:
Yeah, I've struggled w/ deciphering what's in-scope of this ticket -- I think it's definitely inflated greater than its original scope and greater than its 5 pointing.

I was maybe hoping that the previous big change ticket around better structuring this repo would have dealt with a lot of these niggly changes -- but perhaps its scope was function/file structure rather than content idk

Reply:
I've updated them so the names are clearer what they are checking and also they not check for behaviours not logs.